### PR TITLE
Fix for issue 11123.  Do not return metrics when a config record is requested.

### PR DESCRIPTION
**Pull request by brbzull0**
fixes https://github.com/apache/trafficserver/issues/11123

### CODE REVIEW

File: src/records/RecCore.cc [hunk 1]
```diff
@@ -560,18 +560,20 @@ RecLookupMatchingRecords(unsigned rec_type, const char *match, void (*callback)(
...
-
-  tmp.rec_type  = RECT_ALL;
-  tmp.data_type = RECD_INT;
-
-  for (auto &&[name, val] : ts::Metrics::instance()) {
-    if (regex.match(name.data()) >= 0) {
-      tmp.name         = name.data();
-      tmp.data.rec_int = val;
-      callback(&tmp, data);
+  if (REC_TYPE_IS_STAT(rec_type)) {
```
Comment (brbzull0) [inline src/records/RecCore.cc:563]: This part only cares about the stats.

File: src/records/RecCore.cc [hunk 2]
```diff
@@ -560,18 +560,20 @@ RecLookupMatchingRecords(unsigned rec_type, const char *match, void (*callback)(
...
+    tmp.data_type = RECD_INT;
+
+    for (auto &&[name, val] : ts::Metrics::instance()) {
+      if (regex.match(name.data()) >= 0) {
+        tmp.name         = name.data();
+        tmp.data.rec_int = val;
+        callback(&tmp, data);
+      }
     }
   }
```
Comment (brbzull0) [inline src/records/RecCore.cc:578]: all metrics will be handled here, so maybe we can just return from here and not lookup on the `g_records` which seems to hold only configs.

Comment (masaori335) [reply src/records/RecCore.cc:578]: Sounds reasonable for metrics lookup.

**Review by zwoop — APPROVED** [review]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**